### DISCUSSION TITLE
The bond direction is not showing as expected.

### DISCUSSION THREAD

**Discussion by shuan4638** [rdkit/rdkit#3928] [Q&A] [answered]
As mentioned in the topic, the bond direction is not showing as expected when I call `bond.GetBondDir()` funciton.

Here is an example of me printing out the bond direction of Remdesivir, where the bond 19, 24, 25, 26, 28 should give specific bond direction due to the chirality of the connecting atom. But the output were all None.

```
smi = 'CCC(COC(=O)[C@@H](NP(=O)(Oc1ccccc1)OC[C@H]1O[C@@]([C@@H]([C@@H]1O)O)(C#N)c1ccc2n1ncnc2N)C)CC' # Remdesivir
mol = Chem.MolFromSmiles(smi)
print ('Printing bond directions....')
for bond in mol.GetBonds():
    if str(bond.GetBondDir()) != 'NONE': print (bond.GetIdx(), bond.GetBondDir()) 
```

What I expected is 
```
Printing bond directions....
19 BEGINDASH
24 BEGINWEDGE
25 BEGINWEDGE
26 BEGINWEDGE
28 BEGINWEDGE
```

But I got
```
Printing bond directions....
```

Does anyone know why the directions were not printed?

**Comment by greglandrum** [comment, marked as answer]
The RDKit doesn't generate/store bond wedging information by default. That information is not necessary for the way the RDKit represents chirality.

If you need that information, you can make sure that the molecule has a conformer and then you can generate wedging information:
```
In [13]: from rdkit import Chem

In [14]: from rdkit.Chem import rdDepictor

In [15]: smi = 'CCC(COC(=O)[C@@H](NP(=O)(Oc1ccccc1)OC[C@H]1O[C@@]([C@@H]([C@@H]1O)O)(C#N)c1ccc2n1ncnc2N)C)CC' # Remdesivir
    ...: mol = Chem.MolFromSmiles(smi)
    ...: if  not mol.GetNumConformers():
    ...:     rdDepictor.Compute2DCoords(mol)
    ...: conf = mol.GetConformer()
    ...: Chem.WedgeMolBonds(mol,conf)
    ...: print ('Printing bond directions....')
    ...: for bond in mol.GetBonds():
    ...:     if bond.GetBondDir() != Chem.BondDir.NONE: print (bond.GetIdx(), bond.GetBondDir())
    ...: 
Printing bond directions....
19 BEGINDASH
24 BEGINWEDGE
25 BEGINWEDGE
26 BEGINWEDGE
38 BEGINWEDGE
```
Why do you need bond wedging information?

**Reply by shuan4638** [reply]
Thank you so much Greg! It works as I expected now.

**Reply by shuan4638** [reply]
Because I am trying to build a system for optical chemical structure recognition (OCSR) tool. It needs to recognize the stereo bond as well.
